cmake/maintainer.cmake

```diff
@@ -68,6 +68,14 @@ IF(CMAKE_C_COMPILER_ID STREQUAL "GNU" AND CMAKE_C_COMPILER_VERSION VERSION_LESS
   SET(MY_ERROR_FLAGS ${MY_ERROR_FLAGS} -Wno-error=non-virtual-dtor) # gcc bug 7302
 ENDIF()
 
+FOREACH(LANG C CXX)
+  IF(CMAKE_${LANG}_COMPILER_ID MATCHES "Clang")
+    MY_CHECK_AND_SET_COMPILER_FLAG(-Werror=uninitialized-explicit-init)
+    MY_CHECK_AND_SET_COMPILER_FLAG(-Werror=uninitialized-const-reference)
+    SET(CMAKE_${LANG}_FLAGS "${CMAKE_${LANG}_FLAGS} -Werror=uninitialized")
```

Comment:
Please remove SET(CMAKE_${LANG}_FLAGS "${CMAKE_${LANG}_FLAGS} -Werror=uninitialized")
We already set -Werror, in maintainer mode. We don't need -Werror=unintialized

We should not break the compilation in non maintainer mode just for an uninitialized error like we do not break it for other warnings. We will catch the issue in buildbot if needed.

Reply:
I checked with Serg and he agrees we should not break on errors outside maintainer mode.

Reply:
ok to push after the above is fixed